spec/services/constraint_query_update_service_spec.rb

```diff
@@ -86,4 +94,44 @@
       end
     end
   end
+
+  [ActiveRecord::RecordInvalid, NoMethodError].each do |error|
+    context "when there is an error of type: #{error} creating the constraints" do
+      let!(:local_authority) { create(:local_authority) }
+      let!(:planning_application) { create(:planning_application, local_authority:) }
+
+      before do
+        allow_any_instance_of(Constraint).to receive(:save!).and_raise(error)
```

Comment:
Does this have to be `PlanningApplicationConstraintsQuery` ?

I think just the `ActiveRecord::RecordInvalid` error is rescued too, so no need to test for `NoMethodError`